### PR TITLE
Map and Set

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and commit / comment if anything is wrong

### CODE REVIEW

**Review by always-maap — APPROVED** [review]

File: 1-js/05-data-types/07-map-set/02-filter-anagrams/solution.md [hunk 1]
```diff
@@ -9,14 +9,14 @@ cheaters, hectares, teachers -> aceehrst
...
-We'll use the letter-sorted variants as map keys to store only one value per each key:
+ما از کلمه‌هایی که حروف آنها مرتب شده‌اند به عنوان کلید map استفاده می‌کنیم تا فقط یک مقدار را به ازای یک کلید ذخیره کنیم:
 
 ```js run
 function aclean(arr) {
   let map = new Map();
 
   for (let word of arr) {
-    // split the word by letters, sort them and join back
+    // کلمه را به حروف آن تقسیم می‌کنیم، آنها را مرتب می‌کنیم و به یکدیگر پیوند می‌زنیم
```
Comment (mahdyar) [inline 1-js/05-data-types/07-map-set/02-filter-anagrams/solution.md:19]:
```suggestion
    // کلمه را به حروف آن تقسیم می‌کنیم، آنها را مرتب می‌کنیم و به یکدیگر متصل می‌کنیم
```

File: 1-js/05-data-types/07-map-set/02-filter-anagrams/solution.md [hunk 2]
```diff
@@ -31,9 +31,9 @@ let arr = ["nap", "teachers", "cheaters", "PAN", "ear", "era", "hectares"];
 alert( aclean(arr) );
 ```
 
-Letter-sorting is done by the chain of calls in the line `(*)`.
+ترتیب‌بندی حروف با زنجیره‌ای از فراخوانی‌ها در خط `(*)` انجام می‌شود.
 
-For convenience let's split it into multiple lines:
+برای اینکه بهتر شود بیایید آن را به چند خط تقسیم کنیم:
```
Comment (mahdyar) [inline 1-js/05-data-types/07-map-set/02-filter-anagrams/solution.md:36]:
```suggestion
برای اینکه بهتر شود بیایید آن را به چندین خط تقسیم کنیم:
```

File: 1-js/05-data-types/07-map-set/02-filter-anagrams/task.md [hunk 3]
```diff
@@ -2,27 +2,27 @@ importance: 4
 
 ---
 
-# Filter anagrams
+# واروواژه‌ها را جداسازی کنید
 
-[Anagrams](https://en.wikipedia.org/wiki/Anagram) are words that have the same number of same letters, but in different order.
+[واروواژه‌ها](https://fa.wikipedia.org/wiki/واروواژه) کلمه‌هایی هستند که تعداد برابری از حروف یکسان دارند اما با ترتیبی متفاوت.
```
Comment (mahdyar) [inline 1-js/05-data-types/07-map-set/02-filter-anagrams/task.md:7]:
```suggestion
[واروواژه‌ها](https://fa.wikipedia.org/wiki/واروواژه) کلمه‌هایی هستند که تعداد برابری از حروف یکسان دارند، اما با ترتیبی متفاوت.
```

File: 1-js/05-data-types/07-map-set/02-filter-anagrams/task.md [hunk 4]
```diff
@@ -2,27 +2,27 @@ importance: 4
...
+برای مثال:
 
 ```
 nap - pan
 ear - are - era
 cheaters - hectares - teachers
 ```
 
-Write a function `aclean(arr)` that returns an array cleaned from anagrams.
+یک تابع `aclean(arr)` بنویسید که آرایه‌ای پاک شده از واروواژه‌ها را برمی‌گرداند.
```
Comment (mahdyar) [inline 1-js/05-data-types/07-map-set/02-filter-anagrams/task.md:17]:
```suggestion
یک تابع `aclean(arr)` بنویسید که آرایه‌ای تهی از واروواژه‌ها را برمی‌گرداند.
```

File: 1-js/05-data-types/07-map-set/article.md [hunk 5]
```diff
@@ -118,41 +118,41 @@ let recipeMap = new Map([
...
 ]);
 
-// iterate over keys (vegetables)
+// در کلیدها حلقه بزن (سبزیجات)
 for (let vegetable of recipeMap.keys()) {
   alert(vegetable); // cucumber, tomatoes, onion
 }
 
-// iterate over values (amounts)
+// در مقدارها حلقه بزن (میزان آنها)
```
Comment (mahdyar) [inline 1-js/05-data-types/07-map-set/article.md:126]:
```suggestion
// در مقدارها حلقه بزن (میزان آن‌ها)
```

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Comment by mahdiHashemi14** [comment]
Thank you, @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHashemi14!

**Pull request merged**